types/lru-cache/index.d.ts

```diff
@@ -97,54 +98,79 @@ declare class LRUCache<K, V> {
     rforEach<T = this>(callbackFn: (this: T, value: V, key: K, cache: this) => void, thisArg?: T): void;
 
     /**
-     * Return an array of the keys in the cache.
+     * Return a generator yielding the keys in the cache.
+     */
+    keys(): Generator<K>;
+
+    /**
+     * Return a generator yielding [key, value] pairs.
+     */
+    values(): Generator<V>;
+
+    /**
+     * Return an array of the entries in the cache.
      */
-    keys(): K[];
+    entries(): Generator<[K, V]>;
+
+    [Symbol.iterator](): Iterator<[K, V]>;
 
     /**
-     * Return an array of the values in the cache.
+     * Return an array of [key, entry] objects which can be passed to cache.load()
      */
-    values(): V[];
+    dump(): Array<[K, LRUCache.Entry<V>]>;
 
     /**
-     * Return an array of the cache entries ready for serialization and usage with `destinationCache.load(arr)`.
+     * Reset the cache and load in the items in entries in the order listed.
+     * Note that the shape of the resulting cache may be different if the
+     * same options are not used in both caches.
      */
-    dump(): Array<LRUCache.Entry<K, V>>;
+    load(cacheEntries: ReadonlyArray<[K, LRUCache.Entry<V>]>): void;
 
     /**
-     * Loads another cache entries array, obtained with `sourceCache.dump()`,
-     * into the cache. The destination cache is reset before loading new entries
-     *
-     * @param cacheEntries Obtained from `sourceCache.dump()`
+     * Evict the least recently used item, returning its value or `undefined` if cache is empty.
      */
-    load(cacheEntries: ReadonlyArray<LRUCache.Entry<K, V>>): void;
+    pop(): V | undefined;
 }
 
 declare namespace LRUCache {
+    type DisposeReason = "evict" | "set" | "delete";
+
+    type SizeCalculator<K, V> = (value: V, key: K) => number;
+
+    type Disposer<K, V> = (value: V, key: K, reason: DisposeReason) => void;
+
     interface Options<K, V> {
+        /** @deprecated */
+        length?: SizeCalculator<K, V>;
+        /** @deprecated */
+        maxAge?: number;
```

Comment:
```ts
        /**
         * Maximum age in ms. Items are not pro-actively pruned out as they age,
         * but if you try to get an item that is too old, it'll drop it and return
         * undefined instead of giving it to you.
         *
         * @deprecated use options.ttl instead
         * @since 7.0.0
         */
        maxAge?: number;
```